File: src/org/baksia/rustycage/wizards/RustProjectWizard.java
package org.baksia.rustycage.wizards;

import org.eclipse.jface.viewers.ISelection;
import org.eclipse.jface.viewers.IStructuredSelection;
import org.eclipse.jface.wizard.Wizard;
import org.eclipse.ui.INewWizard;
import org.eclipse.ui.IWorkbench;

public class RustProjectWizard extends Wizard implements INewWizard {

    private RustProjectPage rustProjectPage;
    private ISelection selection;


    public RustProjectWizard() {
        setNeedsProgressMonitor(true);
    }

    @Override
    public void addPages() {
        rustProjectPage = new RustProjectPage(selection);
        addPage(rustProjectPage);
    }

    @Override
    public void init(IWorkbench workbench, IStructuredSelection selection) {
        this.selection = selection;
    }

    @Override
    public boolean performFinish() {
        return rustProjectPage.createProject();
    }

}

File: rustycage-plugin/org/baksia/rustycage/wizards/RustProjectWizard.scala
package org.baksia.rustycage.wizards

import org.eclipse.jface.wizard.Wizard
import org.eclipse.jface.dialogs.IDialogSettings
import org.eclipse.jface.viewers.IStructuredSelection
import org.eclipse.ui.INewWizard
import org.eclipse.swt.graphics.Image
import org.eclipse.swt.widgets.Composite
import org.eclipse.ui.IWorkbench
import org.eclipse.jface.viewers.ISelection

class RustProjectWizard extends Wizard with INewWizard {
  setNeedsProgressMonitor(true)

  override def addPages() {
    rustProjectPage = new RustProjectPage(selection)
    addPage(rustProjectPage)
  }

  override def init(workbench: IWorkbench, selection: IStructuredSelection) {
    RustProjectWizard.this.selection = selection
  }

  override def performFinish(): Boolean = rustProjectPage.createProject()

  private var rustProjectPage: RustProjectPage = _
  private var selection: ISelection = _
}